### PR TITLE
[Feature] [connectors] Add spark cassandra source

### PR DESCRIPTION
**Pull request by zixi0825**
## Purpose of this pull request

This pull request adds spark cassandra source. #985

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors/seatunnel-connector-spark-cassandra/src/main/scala/org/apache/seatunnel/spark/source/Cassandra.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        reader.options(optionMap)
+      case Failure(exception) => // do nothing
+    }
+
+    reader.load()
+  }
+
+  override def checkConfig(): CheckResult = {
+    check(config, "table", "keyspace")
+  }
```
Comment (wuchunfu) [inline seatunnel-connectors/seatunnel-connector-spark-cassandra/src/main/scala/org/apache/seatunnel/spark/source/Cassandra.scala:59]: We have now changed `check` to `checkAllExists`, please check, thanks

File: docs/en/spark/configuration/source-plugins/Cassandra.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+### Options
+
+| name             | type   | required | default value |
+| --------------   | ------ | -------- | ------------- |
+| [table](#table-string)            | string | yes      | -             |
+| [keyspace](#keyspace-string)         | string | yes      | -             |
+| [cluster](#cluster-string)          | string | no       | default       |
+| [pushdown](#pushdown-string)         | string | no       | true         |
+
```
Comment (wuchunfu) [inline docs/en/spark/configuration/source-plugins/Cassandra.md:28]: If I remember correctly `result_table_name` seems to be required, please check, thanks

**Comment by CalvinKirs** [comment]
I'll close it due to no feedback for a long time, if you make an update, leave a comment below and I'll reopen it

**Comment by zixi0825** [comment]
@CalvinKirs please reopen it ,thanks

**Pull request closed without merging**